frontend/packages/dev-console/src/components/topology/components/nodes/WorkloadNode.tsx

```diff
@@ -45,114 +42,128 @@ export type WorkloadNodeProps = {
+            kind={workloadData.kind}
+            element={element}
+            dropTarget={dropTarget}
+            canDrop={canDrop}
+            {...rest}
+            attachments={[
+              repoIcon && (
+                <Tooltip key="edit" content="Edit Source Code" position={TooltipPosition.right}>
+                  <Decorator
+                    x={cx + radius - decoratorRadius * 0.7}
+                    y={cy + radius - decoratorRadius * 0.7}
+                    radius={decoratorRadius}
+                    href={editUrl}
+                    external
+                  >
+                    <g transform={`translate(-${decoratorRadius / 2}, -${decoratorRadius / 2})`}>
+                      {repoIcon}
+                    </g>
+                  </Decorator>
+                </Tooltip>
+              ),
+              url && (
+                <Tooltip key="route" content="Open URL" position={TooltipPosition.right}>
+                  <Decorator
+                    x={cx + radius - decoratorRadius * 0.7}
+                    y={cy + -radius + decoratorRadius * 0.7}
+                    radius={decoratorRadius}
+                    href={url}
+                    external
+                    circleRef={urlAnchorRef}
+                  >
+                    <g transform={`translate(-${decoratorRadius / 2}, -${decoratorRadius / 2})`}>
+                      <ExternalLinkAltIcon style={{ fontSize: decoratorRadius }} title="Open URL" />
+                    </g>
+                  </Decorator>
+                </Tooltip>
+              ),
+              <BuildDecorator
+                resource={resourceObj}
+                key="build"
+                workloadData={workloadData}
+                x={cx - radius + decoratorRadius * 0.7}
+                y={cy + radius - decoratorRadius * 0.7}
+                radius={decoratorRadius}
+              />,
+              <MonitoringAlertsDecorator
+                key="monitoringAlert"
+                monitoringAlerts={workloadData.monitoringAlerts}
+                workload={element}
+                x={cx - radius + decoratorRadius * 0.7}
+                y={cy - radius + decoratorRadius * 0.7}
+                radius={decoratorRadius}
+              />,
+            ]}
+          >
+            {donutStatus ? (
+              <PodSet size={size} x={cx} y={cy} data={donutStatus} showPodCount={showPodCount} />
+            ) : null}
+          </BaseNode>
+        </Tooltip>
+      </g>
+    );
+  },
+);
+
+const WorkloadNode: React.FC<WorkloadNodeProps> = observer(({ element, ...rest }) => {
+  const resource = getTopologyResourceObject(element.getData());
+  const { podData, loadError, loaded } = usePodsWatcher(
+    resource,
+    resource.kind,
+    resource.metadata.namespace,
   );
-};
+  const donutStatus: PodRCData = React.useMemo(() => {
+    if (!loadError && loaded) {
+      return podData;
+    }
+    return null;
+  }, [loadError, loaded, podData]);
```

Comment:
You don't need `useMemo` here because there isn't savings on efficiency:
```suggestion
  const donutStatus: PodRCData = !loadError && loaded ? podData : null;
```